### ISSUE TITLE
Check whether variables are optimized and gradients are set to zero

### ISSUE THREAD

**Issue by rickstaa** [rickstaa/LAC-TF2-TORCH-translation#3]
# Report

While checking this, I found an error in the translation on L33-L35 of the `pytorch_l.py` script:

https://github.com/rickstaa/filter_LAC_tf_rewrite/blob/b4e2c3103c227fb09f7037d381fb2dc42b983904/LAC/pytorch_l.py#L33-L35

This should have been wrapped with the `nn.parameter` class:

https://github.com/rickstaa/filter_LAC_tf_rewrite/blob/01254b839929efc900bc1c9a4ad435e06263670b/LAC/pytorch_l.py#L33-L35

Otherwise, the optimizer doesn't know about the existence when using the `model.parameters()` method. Consequently, the weights and biases of the first layer are not updated during the optimisation. Further, also their gradients are not set to zero when calling the `.zero_grad()` method.

https://github.com/rickstaa/filter_LAC_tf_rewrite/blob/01254b839929efc900bc1c9a4ad435e06263670b/LAC/LAC_V1_translated.py#L172

# NOTE: 

This error was introduced because I tried to do a one-to-one translation. Since Pytorch doesn't use a static graph, you must define the weights as a parameter if you want to use the  `model.parameters()` method. If you do not define them as parameters, you supply them to the optimizer yourself. This bug is not present in the [SpinningUp](https://github.com/rickstaa/machine_learning_control/blob/0591e07c726f454a129dbd66fb3136cb1130e88a/machine_learning_control/control/utils/helpers.py#L11) translation since there the [nn.Sequential](https://pytorch.org/docs/stable/generated/torch.nn.Sequential.html) method is used to create the network:

```python
    layers = []
    for j in range(len(sizes) - 1):
        act = activation if j < len(sizes) - 2 else output_activation
        layers += [nn.Linear(sizes[j], sizes[j + 1]), act()]
    return nn.Sequential(*layers
```
